### PR TITLE
Add Additional Topics Under String Section✨

### LINKED ISSUE #1878: [Idea]: Add Additional Topics Under String Section
### Idea Title

Add Additional Topics Under String Section

### Idea Description

To enhance the coverage of string manipulation and operations, the following additional topics should be added to the String Section. This will help users deepen their understanding of working with strings in various scenarios.
- String Basics:
- String Operations 

### Potential Benefits

- [ ] Improves user experience
- [ ] Enhances learning
- [X] Adds a new feature
- [ ] Improves code quality
- [ ] Other

### Implementation Suggestions (Optional)

_No response_

### PR DESCRIPTION
**Pull request by AnushkaChouhan25**
### Description
- String Basics
- String Operations 

Fixes #1878 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Strings/String Basics.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
+---
+id: String-basics
```
Comment (ajay-dhangar) [inline docs/Strings/String Basics.md]: replace `id: String-basics`1 with `id: string-basics`

File: docs/Strings/String Operation.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
+---
+id: String-operations
```
Comment (ajay-dhangar) [inline docs/Strings/String Operation.md]: replace `id: String-operations` with `id: string-operations`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by AnushkaChouhan25** [comment]
@ajay-dhangar DONE

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**